### PR TITLE
Makes used grenades slightly more obvious. Also prevents DOSing admins.

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
requested here by ike https://github.com/BeeStation/BeeStation-Hornet/projects/2#card-77266273 (i had nothing to do so i checked what is there to do)
Port of https://github.com/tgstation/tgstation/pull/64643 and https://github.com/tgstation/tgstation/pull/64730
Adds a variable to track whether a grenade has already been used.
If it has it prevents the grenade from detonating again until the grenade has been reset. (Resetting only implemented for chemical grenades.)
It also prevents priming the grenade/pulsing the grenades wires from messaging admins (but still lets them log) because there's no point in telling the admins if the grenade isn't going to do anything.
The status of the grenade shows up in the examine text.

## Why It's Good For The Game

1. Makes it slightly more obvious that the cleaner grenades in the halls have already been used.
2. Prevents DOSing admins with grenade logging spam.

## Changelog
:cl:
add: Used grenades now look used when examined.
fix: It should no longer be possible to DC admins by spamming grenades.
/:cl:

### CODE REVIEW

**Comment by KubeRoot** [comment]
> If it has it prevents the grenade from detonating again until the grenade has been reset. (Resetting only implemented for chemical grenades.)

Seems to me like you didn't port the resetting here?

**Comment by Dejaku51** [comment]
woops, i'll check everything again

**Comment by Dejaku51** [comment]
Done

File: code/game/objects/items/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -61,6 +61,11 @@
 	if(istype(I,/obj/item/assembly) && stage == GRENADE_WIRED)
 		wires.interact(user)
 	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		if(dud_flags & GRENADE_USED)
+			if (do_after(user, 2 SECONDS, src))
+				to_chat(user, "<span class='notice'>You reset the trigger</span>")
```
Comment (mcmeiler) [inline code/game/objects/items/grenades/chem_grenade.dm]:
```suggestion
				to_chat(user, "<span class='notice'>You reset the trigger.</span>")
```
Very minor typo, missing a period.

**Review by mcmeiler — CHANGES_REQUESTED** [review]

**Review by Archanial — APPROVED** [review]

**Pull request merged**